HafmanWorker.cs

```diff
@@ -0,0 +1,136 @@
+using Hafman;
+
+namespace Havman;
+
+public class HafmanWorker
+{
+	private Dictionary<string, string> codeDict { get; }
+	private Dictionary<string, string> deCodeDict { get; }
+	private List<string> lines { get; }
+
+	public HafmanWorker(Node root, List<string> line)
+	{
+		lines = line;
+		codeDict = new Dictionary<string, string>();
+		deCodeDict = new Dictionary<string, string>();
+		var path = "";
+		while (path != " ")
+		{
+			var newLeaf = findLeaf(root);
+			if (newLeaf.Item2.Length <= 1)
+			{
+				codeDict.Add(newLeaf.Item2, newLeaf.Item1);
+				deCodeDict.Add(newLeaf.Item1, newLeaf.Item2);
+			}
+
+			if (newLeaf.Item1 == "")
+			{
+				path = " ";
+			}
+		}
+	}
+
+	public Tuple<string, string> findLeaf(Node root)
+	{
+		string path = "";
+		var current = root;
+		while (true)
+		{
+			if (current.LeftNode == null)
+			{
+				if (current.RightNode == null)
+				{
+					break;
+				}
+			}
+
+			if (current.LeftNode != null)
+			{
+				current = current.LeftNode;
+				path += 0;
+			}
+			else if (current.RightNode != null)
+			{
+				current = current.RightNode;
+				path += 1;
+			}
+		}
+
+		var ans = new Tuple<string, string>(path, current.Data);
+		if (path == "") return ans;
+		current.setNull();
+		return ans;
+	}
+
+	public void WorkerIncode()
+	{
+		var newCrypt = "";
+		var pathd = @"ans";
+		newCrypt += deCodeDict.Count + Environment.NewLine;
+		newCrypt = deCodeDict.Aggregate(newCrypt,
+			(current, VARIABLE) => current + (VARIABLE.Key + " " + VARIABLE.Value + Environment.NewLine));
+		var liner = lines.Aggregate("", (current, line) => current + line);
+		var tokens = new Tokenizer().Tokenize(liner);
+		var wordCrypt = "";
+		foreach (var token in tokens)
+		{
+			wordCrypt += codeDict[token];
+		}
+
+		wordCrypt = Convert.ToString(Convert.ToInt64(wordCrypt, 2), 10);
+		wordCrypt += Environment.NewLine;
+		newCrypt += wordCrypt;
+		File.WriteAllText(pathd, newCrypt);
+	}
+
+	public void WorkerDecode()
+	{
```

Comment:
The method would rather return decoded text string and take the encoded text as a parameter.
So:
1. Separate method/class reads text from encoded file to string variable 
2. String variable's passed into `public string DecodeText(string encodedText)`  method 
3. Method returns plain text that's stored in a string variable and could be written in a console by something other than decoder class